### PR TITLE
[PLAT-451] Mint optimisation

### PR DESCRIPTION
**Pull request by DidacSF**
## Description

See https://ajunanetwork.atlassian.net/browse/PLAT-451 for details. In short:

* Reduced the amount of random numbers generated on the minting process.

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [ ] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [x] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [ ] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [x] Tests for the changes have been added
- [x] Necessary documentation is added (if appropriate)
- [x] Formatted with `cargo fmt --all`
- [x] Linted with `cargo clippy --all-features --all-targets`
- [x] Tested with `cargo test --workspace --all-features --all-targets`

### CODE REVIEW

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 1]
```diff
@@ -346,17 +354,19 @@ pub mod pallet {
 			Ok(season)
 		}
 
-		fn random_number(who: &T::AccountId, until: u8) -> u8 {
+		#[inline]
```
Comment (cowboy-bebug) [inline pallets/ajuna-awesome-avatars/src/lib.rs:357]: nice :+1:

File: pallets/ajuna-awesome-avatars/src/lib.rs [hunk 2]
```diff
@@ -369,17 +379,19 @@ pub mod pallet {
 				}
 				random_tier
 			};
-			let random_variation = Self::random_number(who, season.max_variations);
+			let random_variation = hash[index + 1] % MAX_PERCENTAGE;
```
Comment (cowboy-bebug) [inline pallets/ajuna-awesome-avatars/src/lib.rs:382]: Opps I think this needs a change :)

**Review by cowboy-bebug — CHANGES_REQUESTED** [review]
Apart from one change, LGTM! Nice work :+1:

File: pallets/ajuna-awesome-avatars/src/lib.rs (hunk 2 above)
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/lib.rs]: Yeah, I see the issue, changed!

File: pallets/ajuna-awesome-avatars/src/lib.rs (hunk 1 above)
Comment (DidacSF) [reply pallets/ajuna-awesome-avatars/src/lib.rs:357]: :fire:

**Review by cowboy-bebug — APPROVED** [review]
LGTM

**Pull request merged**